data/data/azure/bootstrap/main.tf

```diff
@@ -200,17 +200,19 @@ resource "azurerm_linux_virtual_machine" "bootstrap" {
   # included here because it is required by the Azure ARM API.
   admin_password                  = "NotActuallyApplied!"
   disable_password_authentication = false
+  encryption_at_host_enabled      = var.azure_master_encryption_at_host_enabled
 
   identity {
     type         = "UserAssigned"
     identity_ids = [var.identity]
   }
 
   os_disk {
-    name                 = "${var.cluster_id}-bootstrap_OSDisk" # os disk name needs to match cluster-api convention
-    caching              = "ReadWrite"
-    storage_account_type = var.azure_master_root_volume_type
-    disk_size_gb         = 100
+    name                   = "${var.cluster_id}-bootstrap_OSDisk" # os disk name needs to match cluster-api convention
+    caching                = "ReadWrite"
+    storage_account_type   = var.azure_master_root_volume_type
+    disk_size_gb           = 100
+    disk_encryption_set_id = var.azure_master_disk_encryption_set_id == "" ? null : var.azure_master_disk_encryption_set_id
```

Comment:
I steered you wrong here: the way you had it before, with a default of `null` is much better. I was under the impression that an empty string would be the equivalent of null, but I was mistaken. Sorry about that.